src/basho_bench_worker.erl

```diff
@@ -334,3 +349,27 @@ rate_worker_run_loop(State, Lambda) ->
         ExitReason ->
             exit(ExitReason)
     end.
+
+%% Not passing state API - expect non-list spec or error during new attempt
+init_generators(false, Config, Id, Module) ->
+    Module:new(Config, Id);
+%% Passing state API - process list or turn single-spec into [{default,Generator}]
+init_generators(true, ConfigList, Id, Module) -> 
+    config_to_generators(ConfigList, Id, Module, []).
+
+config_to_generators([], _Id, _Module, ACC) ->
+    ACC;
+config_to_generators([{Name, Spec}|ConfigListTail], Id, Module, ACC) ->
+    config_to_generators(ConfigListTail, Id, Module, [ {Name, Module:new(Spec, Id)} | ACC]);
+config_to_generators(Spec, Id, Module, _ACC) when is_tuple(Spec) ->
+    [{default, Module:new(Spec, Id)}].
```

Comment:
I'm still not a huge fan of implicitly adding the notion of a default generator. Seems like something the user should specify explicitly themselves. That would also simplify this logic a lot.

I would probably combine the last two functions like so:

```
init_generators(PassState, Configs, Id, Module) ->
    case PassState of
        false ->
            Module:new(Config, Id)
        true ->
            [{N, Module:new(S, Id)} || {N,S} <- Configs]
    end.
```

Or if you want to do the pattern matching in the function heads:
```
init_generators(false, Config, Id, Module) when is_tuple(Config) ->
    Module:new(Config, Id);
init_generators(true, Configs, Id, Module) when is_list(Configs) ->
    [{N, Module:new(S, Id)} || {N,S} <- Configs].
```